vm/systemSmartContracts/delegation.go

```diff
@@ -224,14 +225,30 @@ func (d *delegation) init(args *vmcommon.ContractCallInput) vmcommon.ReturnCode
 		d.eei.AddReturnMessage("invalid number of arguments to init delegation contract")
 		return vmcommon.UserError
 	}
+	//TODO: Shouldn't be done these checks also in the init method?
+	serviceFee := big.NewInt(0).SetBytes(args.Arguments[1]).Uint64()
+	if serviceFee < d.minServiceFee || serviceFee > d.maxServiceFee {
+		d.eei.AddReturnMessage("service fee out of bounds")
+		return vmcommon.UserError
+	}
+	maxDelegationCap := big.NewInt(0).SetBytes(args.Arguments[0])
+	if maxDelegationCap.Cmp(zero) < 0 {
+		d.eei.AddReturnMessage("invalid max delegation cap")
+		return vmcommon.UserError
+	}
+	if args.CallValue.Cmp(zero) < 0 {
+		d.eei.AddReturnMessage("invalid call value")
+		return vmcommon.UserError
+	}
 
 	ownerAddress = args.CallerAddr
+	//TODO: Why is done the following SetStorage with the same key and value -> DelegationSystemSCKey?
```

Comment:
this is needed for the rewards creator to check that reward destination is a delegation system smart contract.

Reply:
ok